src/csharp/SDK/Image.cs

```diff
@@ -583,31 +577,46 @@ protected virtual void Dispose(bool disposing)
                 {
                     if (disposing)
                     {
+                        //this.FlushMemory();
+
+                        // If we have a native buffer wrapper, dispose it since we
+                        // are a wrapper for it's IMemoryOwner<T> interface
+                        this.nativeBufferWrapper?.Dispose();
+
                         // TODO: dispose managed state (managed objects).
+                        Allocator.Singleton.UnregisterForDisposal(this);
+
+                        handle.Close();
+                        handle = null;
                     }
 
-                    handle.Close();
-                    handle = null;
+                    if (this.managedBufferCache != null)
```

Comment:
This should probably be inside the "if (disposing)"

Reply:
No, we need to do this during finalization, otherwise the buffer is leaked since it is referenced by the Allocator's book keeping.

Reply:
Hmm, ok that make sense. However, you closed your handle above, which could have freed the memory right?